droopler.install

```diff
@@ -1378,3 +1380,60 @@ function droopler_update_8115() {
   $updater->checklist()
     ->saveProgress(['v2.2' => $values]);
 }
+
+/**
+ * Droopler 2.2 Migrate column count field.
+ */
+function droopler_update_8116(&$sandbox) {
+  if (!isset($sandbox['total'])) {
+    $query = \Drupal::entityQuery('paragraph')
+      ->exists('field_d_number')
+      ->execute();
+
+    $sandbox['total'] = count($query);
+    $sandbox['current'] = 0;
+  }
+
+  $paragraphs_per_batch = 25;
+
+  $query = \Drupal::entityQuery('paragraph')
+    ->exists('field_d_number')
+    ->range($sandbox['current'], $sandbox['current'] + $paragraphs_per_batch)
+    ->execute();
+
+  foreach ($query as $pid) {
+    $paragraph = \Drupal::entityTypeManager()
+      ->getStorage('paragraph')
+      ->load($pid);
+    $settings = $paragraph->get('field_d_settings') ? $paragraph->get('field_d_settings')->value : "{}";
+
+    $json = json_decode($settings);
+    if (empty($json->column_count)) {
+      $json->column_count = $paragraph->field_d_number ? ($paragraph->field_d_number->value ?? '4') : '4';
+      $paragraph->set('field_d_settings', json_encode($json));
+      $paragraph->save();
+    }
+    $paragraph->save();
+    $sandbox['current']++;
+  }
+
+  \Drupal::messenger()->addMessage($sandbox['current'] . ' paragraphs processed.');
+  $sandbox['#finished'] = $sandbox['total'] == 0 ? 1 : $sandbox['current'] / $sandbox['total'];
```

Comment:
mialem pytac, ale to :point_up: sugeruje, ze dla ludzi, ktorzy nie mieli tych paragrafow dodanych (z polem field_d_number) to tez zadziala?

Reply:
zadziała, po prostu nie zaaktualizuje żadnych paragrafów, jedyne co zrobi to usunie field config od field_d_number